### PR TITLE
A couple of parameters-on-public-dashboards loose ends

### PR DESCRIPTION
**Pull request by rauchy**
## What type of PR is this? (check all applicable)
- [x] Bug Fix

## Description
* Replace "unsafe widget" error messages only for those that have a visualization
* Rethrow the error

## Related Tickets & Documents
Follow up for #3659

### CODE REVIEW

**Comment by ranbena** [comment]
@rauchy I think a better approach is to make sure `load()` consistently returns a promise.

We can return an already resolved promise:
```js
load(force, maxAge) {
  if (!this.visualization) {
     return Promise.resolve();
  }
```

**Comment by rauchy** [comment]
> I think a better approach is to make sure load() consistently returns a promise.

Totally a better approach. Was just afraid of breaking other usages, but everything seems to work.

File: client/app/pages/dashboards/public-dashboard-page.js [hunk 1]
```diff
@@ -44,6 +44,8 @@ const PublicDashboardPage = {
             if (!isSafe) {
               error.errorMessage = 'This query contains potentially unsafe parameters and cannot be executed on a publicly shared dashboard.';
             }
+
+            throw error;
```
Comment (ranbena) [inline client/app/pages/dashboards/public-dashboard-page.js]: Makes sense but does it have any affect on the flow?

Comment (rauchy) [reply client/app/pages/dashboards/public-dashboard-page.js]: It doesn't right now, but as a good practice it's best to re-throw. (You never know if someone would want to handle this in the future)

Comment (arikfr) [reply client/app/pages/dashboards/public-dashboard-page.js]: This won't blow up the dashboard load?

Comment (ranbena) [reply client/app/pages/dashboards/public-dashboard-page.js]: Nope - gets caught by original promise.
https://github.com/getredash/redash/blob/c3cc65a21d74408da490ee076ae34bfd0391f5f6/client/app/services/widget.js#L139-L142

**Pull request merged**